comp/core/configsync/configsyncimpl/sync.go

```diff
@@ -10,12 +10,57 @@ import (
 	"encoding/json"
 	"net/http"
 	"reflect"
+	"strconv"
 	"time"
 
 	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
 	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
 )
 
+func (cs *configSync) updater() {
+	cfg, err := fetchConfig(cs.ctx, cs.client, cs.Authtoken.Get(), cs.url.String())
+	if err != nil {
+		if cs.connected {
+			cs.Log.Warnf("Failed to fetch config from core agent: %v", err)
+			cs.connected = false
+		} else {
+			cs.Log.Debugf("Failed to fetch config from core agent: %v", err)
+		}
+		return
+	}
+
+	if cs.connected {
+		cs.Log.Debug("Succeeded to fetch config from core agent")
+	} else {
+		cs.Log.Info("Succeeded to fetch config from core agent")
+		cs.connected = true
+	}
+
+	for key, value := range cfg {
+		if key == "logs_config.additional_endpoints" {
+			valueMap, ok := value.(map[string]string)
+			if !ok {
+				updateConfig(cs.Config, key, value)
+			}
+			typedValues := map[string]interface{}{}
+			for cfgkey, cfgval := range valueMap {
+				if cfgkey == "is_reliable" {
+					if b, err := strconv.ParseBool(cfgval); err == nil {
+						typedValues[cfgkey] = b
+					} else {
+						typedValues[cfgkey] = cfgval
+					}
+				}
+				updateConfig(cs.Config, key, typedValues)
+			}
+
+		} else {
+			updateConfig(cs.Config, key, value)
+		}
```

Comment:
We can remove the `else` branch if we use the negative assertion for the key value:
```suggestion
		if key != "logs_config.additional_endpoints" {
			updateConfig(cs.Config, key, value)
		}

		valueMap, ok := value.(map[string]string)
		if !ok {
			updateConfig(cs.Config, key, value)
		}
		typedValues := map[string]interface{}{}
		for cfgkey, cfgval := range valueMap {
			if cfgkey == "is_reliable" {
				if b, err := strconv.ParseBool(cfgval); err == nil {
					typedValues[cfgkey] = b
				} else {
					typedValues[cfgkey] = cfgval
				}
			}
			updateConfig(cs.Config, key, typedValues)
		}
```

Reply:
These don't do the same, but if you're concerned with the cyclomatic complexity I will reverse the logic and `continue`.